### PR TITLE
Update core aten decomp table

### PR DESCRIPTION
**Pull request by angelayi**
Updated the decomposition table based on the existing [Core ATen IR](https://pytorch.org/docs/stable/ir.html) list, and moved rest of decompositions to inductor's decomposition table.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_decomp/__init__.py [hunk 1]
```diff
@@ -283,16 +272,8 @@ def core_aten_decompositions() -> Dict[OpOverload, Callable]:
             aten.nan_to_num,
             aten.nan_to_num_,
             aten.narrow,
-            aten.native_batch_norm,
```
Comment (SherlockNoMad) [inline torch/_decomp/__init__.py:286]: The story around batch_norm is a bit complicated... 
Ideally, all variants of batch_norms should be decomp into "the_one_batch_norm"... 
But I am not sure which variant is "the_one_batch_norm"? 

I currently marked "aten.native_batch_norm" as core, but this is not necessarily correct.

**Review by SherlockNoMad — APPROVED** [review]
LGTM, need more discussion on batch_norm.

File: torch/_decomp/__init__.py (hunk 1 above)
Comment (angelayi) [reply torch/_decomp/__init__.py:286]: Ok, I can add native_batch_norm as part of our op discussion list.

File: torch/_decomp/__init__.py [hunk 2]
```diff
@@ -220,9 +219,6 @@ def core_aten_decompositions() -> Dict[OpOverload, Callable]:
             aten.frac,
             aten.frac_,
             aten._fused_moving_avg_obs_fq_helper,
-            aten.gelu,
-            aten.gelu_,
-            aten.gelu_backward,
```
Comment (Chillee) [inline torch/_decomp/__init__.py:225]: I'm not sure why these are removed? I don't see `gelu_backward` in the Core ATen IR: https://pytorch.org/docs/stable/ir.html

Comment (angelayi) [reply torch/_decomp/__init__.py:225]: Oops sorry, I was mistaken about this -- will put the x_ and x_backward ops back.

File: torch/_inductor/decomposition.py [hunk 3]
```diff
@@ -19,15 +19,29 @@
 
 inductor_decompositions = get_decompositions(
     [
+        aten._adaptive_avg_pool2d_backward,
```
Comment (eellison) [inline torch/_inductor/decomposition.py:22]: We need to check that these produce good inductor generated code before adding them here

Comment (angelayi) [reply torch/_inductor/decomposition.py:22]: How do I do that? These were originally in the core aten decomp table so I would expect this diff does a noop for inductor.

Comment (eellison) [reply torch/_inductor/decomposition.py:22]: ah, if they were already being added nevermind

**Comment by angelayi** [comment]
pytorchbot merge -i

**Comment by gsingh-h** [comment]
@SherlockNoMad These ops also need to be marked with tag `core` in [native_functions.yaml ](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml) so that the doc reflects it, right?

**Comment by angelayi** [comment]
@gsingh-h These are already marked with core. This diff was to update the decomposition table so that the ops marked as core do not get decomposed.

**Pull request closed without merging**